templates/terraform/post_create/dialogflow_agent_tier.go.erb

```diff
@@ -0,0 +1,31 @@
+
+// The tier field is eventually consistent, we need to test for consistency before moving on.
+// Otherwise, the user will see diffs on the field.
+if d.HasChange("tier") {
+	readUrl, err := replaceVars(d, config, "{{DialogflowBasePath}}projects/{{project}}/agent")
+	if err != nil {
+		return err
+	}
+
+	incorrect_attempts := 0
+	consistent_threshold := 8
+	for correct_in_row := 0; correct_in_row < consistent_threshold; {
```

Comment:
I think you should look into using Terraform native code for eventually consistent behavior instead of rolling your own. Check out the [retries section](https://www.terraform.io/docs/extend/resources/retries-and-customizable-timeouts.html#statechangeconf) of the Terraform docs. Also we do some non-operation polling in [cloudrun](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/cloudrun_polling.go) which you can take a gander at.